Document existing project automations and plan implementation of new ones (#3042)
WordPress/openverse: documentation/projects/proposals/project_improvement/20230914-project_proposal_project_improvements.md

# Project proposal: Project board improvements

**Author**: @dhruvkb

## Reviewers

> - [ ] TBD

Reviewer: ```suggestion
- [ ] @aetherunbound
```

Author: Recorded approval in 2f61873.

- [ ] TBD

## Project summary

The end-goal of this project is to have a comprehensive set of boards where one
can track all aspects of work being done in Openverse. This includes issues, PRs
and project discussions. These boards should have lots of different views, for
different purposes, and these views should be standardised and documented.

Each of these projects should be highly automated, where minimal effort is
needed to keep these boards synced to reality. Most human actions should be
complemented by automations that can sync the effects of the event to the board.
These should be a combination of built-in and custom workflows, where custom
workflows are consistent, well-tested and maintainable.

## Goals

> The project both

Reviewer: ```suggestion
This project both:
```

Author: Fixed in 2f61873.

- enhances our productivity by reducing a lot of work required to keep these
  projects up to date
- increases our accountability by helping us stay on top of our work and also
  being aware of what others in the team are working on

## Requirements and success

The following are requirements & success criteria for the project automations:

- all issues and PRs are automatically added to the relevant project boards
- all cards are in the right columns as per their status

> - rarely, if ever, do we need to manually move cards to the right columns

Reviewer: Save for backlog -> TODO!

- all custom automations are in a single place and are easy to maintain
- all automations are documented

The following are requirements & success criteria for the project views:

- views are created based on clear utility in helping maintainers track work
- views are clearly labelled and it's known which view to use for what purpose
- all views and their utility is documented

## Participants

The boards are primarily an internal tool for maintainers and to some extent, an
external tools for contributors (to find potential inroads to contribute) and

> managers (to gauge activity and progress).

Reviewer: ```suggestion
project leadership (to gauge activity and progress).
```

Author: Reworded in 2f61873.

### Expectation for maintainers

- Knowing what issues everyone, including themselves, is working on at a given
  time.

> - Having a clear idea of what issues needs to be worked on after the current

Reviewer: ```suggestion
- Having a clear idea of what issues need to be worked on after the current
```

Author: Reworded in 2f61873.

  issues on hand are resolved.
- Being able to find critical or high priority issues that haven't been
  addressed.
- Knowing the status of their own PRs.

> - Identifying PRs that need reviews to be merged.

Reviewer: ```suggestion
- Identifying PRs that require additional reviews.
```

Author: Fixed in 2f61873.

### Expectation for contributors

- Finding help wanted or good first issues that are not already assigned or
  being worked on by someone else.
- Knowing that their PRs will be noticed, reviewed and merged in a timely
  manner.

> ### Expectation for managers
> 
> - While managers are not directly concerned with the day-to-day issues and PRs,
>   they should be able to see a steady movement of cards from left-to-right.
> - The project discussion boards are more relevant to the management.

Reviewer: ```suggestion
### Expectation for project leadership

- While project leaders are not directly concerned with the day-to-day issues and PRs,
  they should be able to see a steady movement of cards from left-to-right.
- The project discussion boards are more relevant to the leadership.
```

Author: Reworded in 2f61873.

## Miscellaneous

This project should not have any components related to infrastructure, a11y or
marketing as it is internal. Several new workflows will be defined which will be
executed on GitHub Actions.

## Required implementation plans

The following implementation plans would be required:

1. improvements to existing, restoration of previous and creation of new
   automations
   ([plan](/projects/proposals/project_improvement/20230914-implementation_plan_project_automations.md))

> 2. documentation of existing and creation of new useful views for project boards

Reviewer: Would this require a whole implementation plan, or just an issue to document the views themselves?

Author: Just documentation should suffice. We can make views as and when the requirements present themselves but having them documented makes it easy for someone new (or from leadership) to know what views we have and how to best use them to derive insights.
